server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointSnapshotTransferManagerWithoutMetadata.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.index.translog.transfer.FileSnapshot.CheckpointFileSnapshot;
+import org.opensearch.index.translog.transfer.FileSnapshot.TransferFileSnapshot;
+import org.opensearch.index.translog.transfer.FileSnapshot.TranslogFileSnapshot;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Translog transfer manager to transfer {@link TranslogCheckpointSnapshot} by transfering {@link TranslogFileSnapshot} and {@link CheckpointFileSnapshot} separately
+ *
+ * @opensearch.internal
+ */
+public class TranslogCheckpointSnapshotTransferManagerWithoutMetadata implements TranslogCheckpointSnapshotTransferManager {
+
+    private final TransferService transferService;
+    private final FileTransferTracker fileTransferTracker;
+
+    public TranslogCheckpointSnapshotTransferManagerWithoutMetadata(
+        TransferService transferService,
+        FileTransferTracker fileTransferTracker
+    ) {
+        this.transferService = transferService;
+        this.fileTransferTracker = fileTransferTracker;
+    }
+
+    @Override
+    public void transferTranslogCheckpointSnapshot(
```

Comment:
Lets add multiple test cases for this method and ensure that the latch is always counted down in a fixed time.